sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -579,6 +579,32 @@ private[lf] object SBuiltinFun {
     }
   }
 
+  final case object SBTextToContractId extends SBuiltinFun(1) {
+    override private[speedy] def execute[Q](
+        args: util.ArrayList[SValue],
+        machine: Machine[Q],
+    ): Control[Nothing] = {
+      val value = getSText(args, 0)
+      V.ContractId.V1.fromString(value) match {
+        case Right(cid) if !cid.isLocal => Control.Value(SContractId(cid))
+        case Right(_) =>
+          Control.Error(
+            IE.Dev(
+              NameOf.qualifiedNameOfCurrentFunc,
+              IE.Dev.MalformedContractId(value, "unexpected local contract ID"),
+            )
+          )
+        case Left(error) =>
+          Control.Error(
+            IE.Dev(
+              NameOf.qualifiedNameOfCurrentFunc,
+              IE.Dev.MalformedContractId(value, error),
+            )
+          )
```

Comment:
```suggestion
      V.ContractId.V.fromString(value) match {
        case Right(cid) if !cid.isLocal => Control.Value(SContractId(cid))
        case _ =>
          Control.Error(
            IE.Dev(
              NameOf.qualifiedNameOfCurrentFunc,
              IE.Dev.MalformedContractId(value, "cannot parse global contract ID $value"),
            )
          )
        }
```

I think we do not need to give details about why the string cannot be converted to a global contract ID.
- either the user exactly copied an existing persisted cid, 
- or he put rubbish, i.e. the error message will not useful 
- or he tried to be malicious, in this case better to not help me.